docs/sphinx_extensions/dfhack/tool_docs.py

```diff
@@ -0,0 +1,175 @@
+    for keycmd, key, ctx in _KEYBINDS[command]:
+        n = nodes.paragraph()
+        n += nodes.strong('Keybinding:', 'Keybinding:')
+        n += nodes.inline(' ', ' ')
+        for k in key:
+            n += nodes.inline(k, k, classes=['kbd'])
+        if keycmd != command:
+            n += nodes.inline(' -> ', ' -> ')
+            n += nodes.literal(keycmd, keycmd, classes=['guilabel'])
+        if ctx:
+            n += nodes.inline(' in ', ' in ')
+            n += nodes.literal(ctx, ctx)
+        out.append(n)
+    return out
+
+
+def check_missing_keybinds():
+    # FIXME: _KEYBINDS_RENDERED is empty in the parent process under parallel builds
+    # consider moving to a sphinx Domain to solve this properly
+    for missing_command in sorted(set(_KEYBINDS.keys()) - _KEYBINDS_RENDERED):
+        logger.warning('Undocumented keybindings for command: %s', missing_command)
+
+
+# pylint:disable=unused-argument,dangerous-default-value,too-many-arguments
+def dfhack_keybind_role(role, rawtext, text, lineno, inliner,
+                        options={}, content=[]):
+    """Custom role parser for DFHack default keybinds."""
+    return render_dfhack_keybind(text), []
+
+
+class DFHackToolDirectiveBase(sphinx.directives.ObjectDescription):
+    has_content = False
+    required_arguments = 0
+    optional_arguments = 1
+
+    def get_name_or_docname(self):
+        if self.arguments:
+            return self.arguments[0]
+        else:
+            return self.env.docname.split('/')[-1]
+
+    @staticmethod
+    def make_labeled_paragraph(label, content, label_class=nodes.strong, content_class=nodes.inline) -> nodes.paragraph:
+        return nodes.paragraph('', '', *[
+            label_class('', '{}: '.format(label)),
+            content_class('', content),
+        ])
+
+    @staticmethod
+    def wrap_box(*children: List[nodes.Node]) -> nodes.Admonition:
+        return nodes.admonition('', *children, classes=['dfhack-tool-summary'])
+
+    def render_content(self) -> List[nodes.Node]:
+        raise NotImplementedError
+
+    def run(self):
+        return [self.wrap_box(*self.render_content())]
+
+
+class DFHackToolDirective(DFHackToolDirectiveBase):
+    option_spec = {
+        'tags': dfhack.util.directive_arg_str_list,
+        'no-command': rst_directives.flag,
+    }
+
+    def render_content(self) -> List[nodes.Node]:
+        tag_nodes = [nodes.strong(text='Tags: ')]
+        for tag in self.options.get('tags', []):
+            tag_nodes += [
+                addnodes.pending_xref(tag, nodes.inline(text=tag), **{
+                    'reftype': 'ref',
+                    'refdomain': 'std',
+                    'reftarget': 'tag/' + tag,
+                    'refexplicit': False,
+                    'refwarn': True,
+                }),
+                nodes.inline(text=' | '),
+            ]
+        tag_nodes.pop()
+
+        return [
+            self.make_labeled_paragraph('Tool', self.get_name_or_docname()),
```

Comment:
As mentioned in discord, this will be appearing just under a header of the same name, so it's not clear that we need to output the tool name here again.

Reply:
I'll remove it. I tried adding tool type information automatically based on the document path, but failed because `env.docname` is the including document (/docs/tools/...), not the one being included.

Snippet for posterity:
```python
    def get_type_description(self):
        stem = ''
        docname_parts = self.env.docname.split('/')
        if len(docname_parts) >= 2:
            stem = docname_parts[-2].lower()
        if stem.startswith('plugin'):
            return 'plugin'
        elif stem.startswith('script'):
            return 'script'
        elif stem.startswith('builtin'):
            return 'built-in'
        else:
            return 'unknown type'

```